sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -377,7 +377,8 @@ case class InsertIntoTable(
   }
 
   assert(overwrite || !ifNotExists)
-  override lazy val resolved: Boolean = childrenResolved && expectedColumns.forall { expected =>
+  override lazy val resolved: Boolean =
+    childrenResolved && table.resolved && expectedColumns.forall { expected =>
```

Comment:
For `InsertIntoTable`, `table.resolved` is false when `table` is `SimpleCatalogRelation`. If we do not change it, the `resolved` of `InsertIntoTable` will be true. 

In this case, we will not call `CheckAnalysis`, and thus, it will not issue an appropriate exception. That means, this PR requires this change. Let me know if you have any concern. Thanks!